pilot/pkg/networking/plugin/mixer/mixer.go

```diff
@@ -344,16 +347,48 @@ func buildInboundTCPFilter(mesh *meshconfig.MeshConfig, node *model.Proxy, attrs
 	}
 }
 
-func addServiceConfig(node *model.Proxy, filterConfigs map[string]*types.Struct, config *mccpb.ServiceConfig) map[string]*types.Struct {
+func addServiceConfig(outbound bool, push *model.PushContext, hostname model.Hostname, in *plugin.InputParams,
+	filterConfigs map[string]*types.Struct) map[string]*types.Struct {
 	if filterConfigs == nil {
 		filterConfigs = make(map[string]*types.Struct)
 	}
+	var key string
+	var config *mccpb.ServiceConfig
+	if outbound {
+		key = string(hostname) + "outbound"
+	} else {
+		key = string(hostname) + "inbound"
+	}
+	push.Mutex.Lock()
+	filterConfig, ok := push.MixerPerRouteFilterConfig[key]
+	push.Mutex.Unlock()
+	if ok {
+		filterConfigs[mixer] = filterConfig
+		return filterConfigs
+	}
+	if outbound {
+		attrs := addDestinationServiceAttributes(make(attributes), push, hostname)
+		config = &mccpb.ServiceConfig{
+			DisableCheckCalls: disableClientPolicyChecks(in.Env.Mesh, in.Node),
+			MixerAttributes:   &mpb.Attributes{Attributes: attrs},
+			ForwardAttributes: &mpb.Attributes{Attributes: attrs},
+		}
+	} else {
+		config = buildInboundRouteConfig(push, in, in.ServiceInstance)
+	}
 	// These settings are not backward compatible with 0.8.
 	// Proxy version is only available from 1.0 onwards.
```

Comment:
Can we remove this ? 1.1 doesn't need compat with 0.8 ( in separate PR maybe)

Reply:
OK, I will remove this for 1.1 in a new PR. Thanks!